CARL/source/Recognizer.cpp

```diff
@@ -134,104 +143,136 @@ namespace carl::action
                 using SignalT = Signal<gsl::span<const InputSample>>;
```

Comment:
Quite a few bug fixes here; this method was basically rewritten because it didn't work at all before, which makes sense as it was all but untested without a usable integration.